sanic/reloader_helpers.py

```diff
@@ -0,0 +1,121 @@
+import os
+import sys
+import signal
+import subprocess
+from time import sleep
+from multiprocessing import Process
+
+
+def _iter_module_files():
+    """This iterates over all relevant Python files.
+
+    It goes through all
+    loaded files from modules, all files in folders of already loaded modules
+    as well as all files reachable through a package.
+    """
+    # The list call is necessary on Python 3 in case the module
+    # dictionary modifies during iteration.
+    for module in list(sys.modules.values()):
+        if module is None:
+            continue
+        filename = getattr(module, '__file__', None)
+        if filename:
+            old = None
+            while not os.path.isfile(filename):
+                old = filename
+                filename = os.path.dirname(filename)
+                if filename == old:
+                    break
+            else:
+                if filename[-4:] in ('.pyc', '.pyo'):
+                    filename = filename[:-1]
+                yield filename
+
+
+def _get_args_for_reloading():
+    """Returns the executable."""
+    rv = [sys.executable]
+    rv.extend(sys.argv)
+    return rv
+
+
+def restart_with_reloader():
+    """Create a new process and a subprocess in it with the same arguments as
+    this one.
+    """
+    args = _get_args_for_reloading()
+    new_environ = os.environ.copy()
+    new_environ['SANIC_SERVER_RUNNING'] = 'true'
```

Comment:
I think that the name of this env variable is a bit misleading (in my point fo view). for me it sounds like if this variable is set in the server module (or by the server) instead of being set by the reloader, I would suggest renaming it to something as `SANIC_APP_REALOADING` (and changing the logic accordingly).

Hope it helps :)

Reply:
Thanks for your review.
I think this logic is more understandable, Werkzeug uses this way too and I inspired by Werkzeug!

Reply:
I understand and I'm also a big fan of Werkzeug but please evaluate the name one more time :)
And if you can please give me another reason other than werkzeug using it, would be helpful for me then to understand more throughly  :)